### DISCUSSION TITLE
Build Capacitor Apps on a Server

### DISCUSSION THREAD

**Discussion by fisherandrewbiz** [ionic-team/capacitor#3183] [Q&A]
Can anyone point me in the direction of being able to build capacitor apps on a server?  I want to send the server the project files, have it build the Android and iOS projects, and send back the files it generated.

**Comment by IT-MikeS** [comment]
Number of ways to do this, but the easiest way to get this kind of functionality would be to look at AppFlow.

https://ionicframework.com/appflow

---

Past that you need to look into a filesync option across some order of file protocol, such as SMB shares in a local network, and having scripts to trigger build on file changes in the shares. This is a bit of a task to write out in a post however, but I believe in you that you'll figure it out with those concepts I said above 🙂

**Reply by imhoffd** [reply]
AppFlow was made with hybrid apps in mind so it works a lot better with Capacitor. That being said, the `ios/` and `android/` directories of Capacitor apps contain real native apps, so they should be able to be built by any existing system that handles builds native iOS and Android apps, provided you can run additional commands during the build and configure paths.

**Comment by crixx** [comment]
Another option which we use is Visual Studio "Appcenter" https://appcenter.ms , it has an API but don't know if that matches your exact use case.

In our case, we build it on appcenter on every push to GIT and then send the testers a notification with a download link...

Some more info on the exact setup for ios:
- choose the "native Swift" Projekt type. Appcenter will then search for you XCode project file
- add a appcenter-post-clone-hook.sh and run `npm install && npx cap sync` in that hook
- setup a build configuration including the correct certificates and profiles
- setup in the build configuration the "distribute" section
- have fun

Crixx
